src/Microsoft.AspNet.OData.Shared/DefaultODataAPIHandler.cs

```diff
@@ -172,6 +173,36 @@ private TStructuralType GetFilteredItem(IDictionary<string, object> keyValues)
             }
 
             return default(TStructuralType);
-        }  
+        }
+
+        /// <inheritdoc/>
+        public override Task<ODataAPIResponseStatus> TryCreateAsync(IDictionary<string, object> keyValues, out TStructuralType createdObject, out string errorMessage)
+        {
+            return Task.FromResult(TryCreate(keyValues, out createdObject, out errorMessage));
```

Comment:
Given the effort we're putting into removing synchronous I/O from ODL's async API, maybe we should avoid calling synchronous APIs from async code.

If we except the async APIs to be preferred in production, then it would make sense to have an async-first implementation here, where the async methods are truly async and not wrappers around the synchronous APIs. But that does require more work to make sure each call stack performs no synchronous I/O.

Reply:
can we put this effort when we'll be moving this to 8.x?